php/php.blade/src/org/netbeans/modules/php/blade/editor/completion/BladeCompletionProposal.java

```diff
@@ -0,0 +1,469 @@
+        FileObject file = null;
+        if (element != null) {
+            file = element.getFileObject();
+        }
+        if (file != null) {
+            formatter.reset();
+            formatter.appendText(" ");
+            formatter.appendText(file.getName());
+        }
+        return formatter.getText();
+    }
+
+    @Override
+    public ElementKind getKind() {
+        return ElementKind.CONSTRUCTOR;
+    }
+
+    @Override
+    public boolean isSmart() {
+        return true;
+    }
+
+    public static class PhpElementItem extends BladeCompletionProposal {
+
+        public PhpElementItem(ElementHandle element, CompletionRequest request, String previewValue) {
+            super(element, request, previewValue);
+        }
+
+        @Override
+        public String getRhsHtml(HtmlFormatter formatter) {
+            FileObject file = null;
+            if (this.getElement() != null) {
+                file = this.getElement().getFileObject();
+            }
+            if (file != null) {
+                formatter.reset();
+                formatter.appendText(" ");
+                formatter.appendText(file.getNameExt());
+            }
+            return formatter.getText();
+        }
+    }
+
+    public static class NamespaceItem extends PhpElementItem {
+
+        public NamespaceItem(ElementHandle element, CompletionRequest request, String previewValue) {
+            super(element, request, previewValue);
+        }
+
+        @Override
+        public ElementKind getKind() {
+            return ElementKind.PACKAGE;
+        }
+
+        @Override
+        public int getSortPrioOverride() {
+            return -50;//priority
+        }
+    }
+
+    public static class DirectiveItem extends BladeCompletionProposal {
+
+        public DirectiveItem(ElementHandle element, CompletionRequest request, String previewValue) {
+            super(element, request, previewValue);
+        }
+
+    }
+
+    public static class ClassItem extends PhpElementItem {
+
+        protected String namespace = null;
+        
+        public ClassItem(ClassElement element, CompletionRequest request, String previewValue) {
+            super(element, request, previewValue);
+            this.namespace = element.getNamespace();
+        }
+
+        @Override
+        public ElementKind getKind() {
+            return ElementKind.CLASS;
+        }
+
+        @Override
+        public String getRhsHtml(HtmlFormatter formatter) {
+            if (namespace != null && namespace.length() > 0) {
+                return namespace;
+            }
+            return super.getRhsHtml(formatter);
+        }
+
+        @Override
+        public int getSortPrioOverride() {
+            return 10;//priority
+        }
+
+        @Override
+        public String getCustomInsertTemplate() {
+            if (namespace != null && namespace.length() > 0) {
+                return "\\" + namespace + "\\" + element.getName();
```

Comment:
Can change these(`"\\"`) to constants?

Reply:
replaced with EditorStringUtils.NAMESPACE_SEPARATOR